Sources/SentryCrash/Recording/Monitors/SentryCrashMonitor_MachException.c

```diff
@@ -395,6 +395,10 @@ uninstallExceptionHandler()
         g_secondaryPThread = 0;
     }
 
+    // Only this file is calling sentrycrashmc_addReservedThread, so it's fine to clear the list
+    // instead of removing the threads this file added.
+    sentrycrashmc_clearReservedThreads();
```

Comment:
This looks like a dangerous assumption to make. I would move the declaration and definition for this function into this file so it's not exposed to other places that won't know about this comment.

Reply:
Yep, I agree 100%. I'm a bit afraid of refactoring C code 😁, but this one should be easy.